File: attendy-v2/backend/app/schemas/meal.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel

from app.schemas.class_section import ClassSectionOut


class MealRow(BaseModel):
    student_id: uuid.UUID
    name: str
    roll_number: int
    class_section: ClassSectionOut
    status: str  # "present" | "absent"
    event_time: datetime | None
    source: str | None


class MealSheetResponse(BaseModel):
    date: date
    items: list[MealRow]
    total: int
    present_count: int
    absent_count: int

File: attendy-v2/frontend/src/types/meal.ts
import type { ClassSection } from './index'

export interface MealRow {
  student_id: string
  name: string
  roll_number: number
  class_section: ClassSection
  status: 'present' | 'absent'
  event_time: string | null
  source: string | null
}

export interface MealSheetResponse {
  date: string
  items: MealRow[]
  total: number
  present_count: number
  absent_count: number
}
